### PR TITLE
refactor: remove `Validation` from `EntryPoint`

### PR DESCRIPTION
**Pull request by jaschdoc**
Closes https://github.com/flix/flix/issues/8811

### CODE REVIEW

**Comment by magnus-madsen** [comment]
@mlutze This is mostly your code, so I will let you take the first stab.

File: main/src/ca/uwaterloo/flix/language/phase/EntryPoint.scala [hunk 1]
```diff
@@ -133,78 +137,90 @@ object EntryPoint {
...
         // Case 1: Unit -> XYZ. We can ignore the args.
-        case Some(arg) if Scheme.equal(unitSc, Scheme.generalize(Nil, Nil, arg, RigidityEnv.empty), traitEnv, ListMap.empty) =>
-          // TODO ASSOC-TYPES better eqEnv
-          Validation.success(())
+        case Some(arg) if isUnitParameter(traitEnv, unitSc, arg) => // TODO ASSOC-TYPES better eqEnv
 
         // Case 2: Bad arguments. SoftError
-        // Case 3: argVal was None. SoftError
-        case _ => Validation.toSoftFailure((), EntryPointError.IllegalEntryPointArgs(sym, sym.loc))
+        // Case 3: `arg` was None. SoftError
```
Comment (jaschdoc) [inline main/src/ca/uwaterloo/flix/language/phase/EntryPoint.scala:174]: Should be optArg

File: main/src/ca/uwaterloo/flix/language/phase/EntryPoint.scala [hunk 2]
```diff
@@ -133,78 +137,90 @@ object EntryPoint {
...
+        val error = EntryPointError.IllegalEntryPointResult(sym, resultTpe, sym.loc)
+        sctx.errors.add(error)
       }
   }
 
+  /**
+    * Returns `true` iff `declaredEff` is not `Pure`, `IO`, `NonDet` or `Sys`
+    */
+  private def isBadEntryPointEffect(declaredEff: Type) = {
+    declaredEff != Type.Pure && (declaredEff != Type.IO && declaredEff != Type.NonDet && declaredEff != Type.Sys)
```
Comment (mlutze) [inline main/src/ca/uwaterloo/flix/language/phase/EntryPoint.scala:221]: Out of scope for this PR, but this looks like it should be looking for "base effects".

**Review by mlutze — APPROVED** [review]
This phase is so ugly. But you've made it less ugly. Good job.

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**